client-libraries/java/rest-client/src/com/google/android/gcm/server/Message.java

```diff
@@ -61,13 +61,13 @@
   private final String collapseKey;
   private final Boolean delayWhileIdle;
   private final Integer timeToLive;
-  private final Map<String, String> data;
```

Comment:
I think that since the parameter has to be be a string at the time it is sent that it would be better to leave this as a string. Also changing public methods from String to Object could break backwards compatibility.